conjure-go-client/httpclient/response_error_decoder_middleware_test.go

```diff
@@ -79,6 +79,25 @@ func TestErrorDecoderMiddlewares(t *testing.T) {
 				assert.NoError(t, err)
 			},
 		},
+		{
+			name: "307 with relative location",
+			handler: func(w http.ResponseWriter, r *http.Request) {
+				w.Header().Set("Location", "/newPath")
+				w.WriteHeader(307)
+			},
+			verify: func(t *testing.T, u *url.URL, err error) {
+				assert.EqualError(t, err, "httpclient request failed: 307 Temporary Redirect")
+				code, ok := httpclient.StatusCodeFromError(err)
+				assert.True(t, ok)
+				assert.Equal(t, 307, code)
+				location, ok := httpclient.LocationFromError(err)
+				assert.True(t, ok)
+
+				// The redirect url is a path relative to original URI
+				expected := u.ResolveReference(&url.URL{Path: "/newPath"})
```

Comment:
The `location` header returned by the server in this test case only contains a relative path. This relative path should be used instead of the original path in the previous request.